### DISCUSSION TITLE
Gene information SCPA

### DISCUSSION THREAD

**Discussion by Youvika** [jackbibby1/SCPA#24] [Q&A] [answered]
Hi Jack, Awesome package. I love it and planning to use this package in my current project.
Already got some nice results.
But I have a query.
Could we have the gene information also of the perturbed pathways between conditions?
For eg, pathway 1 is changed between condition A Vs Condition B, in X cell type, so which genes are responsible for making that change?
If we get this information I am gonna recommend this amazing tool to my colleagues.
Pardon me if you explained earlier.
Thanks in advance

**Comment by jackbibby1** [comment, marked as answer]
Hi (FYI I've closed the crossposted issue #22 because it's a bit more of a discussion),

Glad it's seeming useful. Because we're using a multivariate analysis in SCPA, it makes it more difficult to define the exact expression change in specific genes that are leading to the signature. I briefly discussed this [here](https://github.com/jackbibby1/SCPA/issues/11#issuecomment-1177829051) at the end of issue #11, and I'd still recommend the same thing i.e. filtering out genes that have little/no expression, and consider the rest of the pathway as a whole -- something like the heatmap in figure 3F our paper [here](https://www.cell.com/cell-reports/pdf/S2211-1247(22)01571-6.pdf). I think considering the pathway like this gives you a more complete overview of the changes between conditions/populations, rather the reducing the pathway change down to a few genes.

Jack

**Reply by Youvika** [reply]
Thanks for your quick reply.
I will try to do it as per your suggestion. I have one more thing to ask.
Hallmark has 50 gene sets and SCPA suggests all 50 hallmark pathways are differentially perturbed in my single-sell dataset. 
I didn't understand this actually. I used "Systems level pathway analysis in disease"  tutorial.
If all the pathways show the change that which one to focus on more?
This is also be seen with KEGG (out of 176/186 pathways) and other gene sets.
Are there more filters that I need to use? (I already used qval >2)
Thanks!!

**Reply by jackbibby1** [reply]
Yup -- if your particular conditions are quite different, then I wouldn't be too surprised that most pathways show some sort of transcriptional change e.g. when we compare Th1 to T central memory cells [here](https://jackbibby1.github.io/SCPA/articles/quick_start.html#plotting-some-basic-output). Rather than considering a particular cut off for the qval, I would consider the pathway changes as graded from small (smaller Qval) to large (larger Qval) and e.g. take the top ranked pathways as your most interesting candidates for follow up work, or look at top pathways that are specific to certain cell types, or top pathways that are shared across many cells types. The `plot_rank()` or `plot_heatmap()` functions mentioned [here](https://jackbibby1.github.io/SCPA/articles/visualisation.html) may also be helpful in visualizing the distribution in the size of pathway changes. Hope that helps

**Reply by Youvika** [reply]
Thanks a lot! This helps:)
